### ISSUE TITLE
[Security] Hardcoded Django Secret Key

### ISSUE THREAD

**Issue by tpansino** [bbilly1/tubearchivist#28]
There is a hardcoded `SECRET_KEY` value [here](https://github.com/bbilly1/tubearchivist/blob/80fcbddd655b8050df4bb16fece5fc1ebc427923/tubearchivist/config/settings.py#L24), in the `settings.py` of the project.

From the [Django docs](https://docs.djangoproject.com/en/3.2/ref/settings/#secret-key) regarding `SECRET_KEY`:

> Warning
> 
> Keep this value secret.
> 
> Running Django with a known SECRET_KEY defeats many of Django’s security protections, and can lead to privilege escalation and remote code execution vulnerabilities.

I recommend changing this line to something like:
```python
SECRET_KEY = str(os.getenv("DJANGO_SECRET_KEY"))
```

Or better yet:
```python
if os.getenv("DJANGO_SECRET_KEY_FILE"):
  open('os.getenv("DJANGO_SECRET_KEY_FILE", "r") as secret_file:
    SECRET_KEY = secret_file.read().strip()
else:
  SECRET_KEY = os.getenv("DJANGO_SECRET_KEY")
```

Which gives the option to pass the value more securely via Docker or Kubernetes Secrets.

(There are also Django settings packages which can reduce the boilerplate code here, provide this kind of file/env var support and do type casting for any setting in Django. I'm out of the loop on Django development these days or I'd recommend one, but I have seen that they exist.)

**Comment by bbilly1** [comment]
Yes, I'm fully aware, and it's not the only security issue: There is no authentication, not all the forms are properly validated, additionally the database connection is open too. And I'm sure there are additional issues. 

I think I'm upfront with that in the [known limitations](https://github.com/bbilly1/tubearchivist#known-limitations). Also check out the roadmap at that opportunity. 

I'm currently still working on some basic functionality and haven't had the chance to look into these things.

Feel free to make pull requests! I'm happy for all contributions.

**Comment by tpansino** [comment]
Hmm. I don't think I would have picked up on those issues just from the Known Limitations. I _do_ think I should have picked up on them from the `0.0.3` version though. :slightly_smiling_face: 

I look forward to the growth of your project! Feel free to close this issue if you wish since you're already aware of it.

**Comment by bbilly1** [comment]
Well, for a version 1.0.0, that would be a very lazy excuse, totally agree. :-) I'll reword the known limitations a little bit to make sure it's clear. 

Hey, any help is welcomed! Do you want to look into making a login system? It's Django, so a lot of heavy lifting is already done.

**Comment by bbilly1** [comment]
Just an update: Most of that is now addressed in [v0.0.7](https://github.com/bbilly1/tubearchivist/releases/tag/v0.0.7). It's not perfect yet, don't put it out into the internet just yet. So I kept the message for now in the know limitations to allow for some more testing. 

Closing this comment.
